plugins/Vectorscope/VectorView.cpp

```diff
@@ -0,0 +1,294 @@
+				const float distanceLog = log10(1 + 9 * abs(distance));
+				const float angleCos = inBuffer[frame][0] / distance;
+				const float angleSin = inBuffer[frame][1] / distance;
+				left  = distanceLog * angleCos * (activeSize - 1) / 4;
+				right = distanceLog * angleSin * (activeSize - 1) / 4;
+			} else {
+				left  = inBuffer[frame][0] * (activeSize - 1) / 4;
+				right = inBuffer[frame][1] * (activeSize - 1) / 4;
+			}
+			x = saturate(right - left + activeSize / 2.f);
+			y = saturate(activeSize - (right + left + activeSize / 2.f));
+			((QRgb*)m_displayBuffer.data())[x + y * activeSize] = m_controls->m_colorFG.rgb();
+		}
+	}
+
+	// Draw background
+	painter.fillRect(displayLeft, displayTop, displayWidth, displayHeight, QColor(0,0,0));
+
+	// Draw the final image
+	QImage temp = QImage(m_displayBuffer.data(),
+						 activeSize,
+						 activeSize,
+						 QImage::Format_RGB32);
+	temp.setDevicePixelRatio(devicePixelRatio());
+	painter.drawImage(displayLeft, displayTop,
+					  temp.scaledToWidth(displayWidth * devicePixelRatio(),
+					  Qt::SmoothTransformation));
+
+	// Draw the grid and labels
+	painter.setPen(QPen(m_controls->m_colorGrid, 1.5, Qt::SolidLine, Qt::RoundCap, Qt::BevelJoin));
+	painter.drawEllipse(QPointF(centerX, centerY), displayWidth / 2.f, displayWidth / 2.f);
+	painter.setPen(QPen(m_controls->m_colorGrid, 1.5, Qt::DotLine, Qt::RoundCap, Qt::BevelJoin));
+	painter.drawLine(QPointF(centerX, centerY), QPointF(displayLeft + gridCorner, displayTop + gridCorner));
+	painter.drawLine(QPointF(centerX, centerY), QPointF(displayRight - gridCorner, displayTop + gridCorner));
+
+	painter.setPen(QPen(m_controls->m_colorLabels, 1, Qt::SolidLine, Qt::RoundCap, Qt::BevelJoin));
+	painter.setFont(boldFont);
+	painter.drawText(displayLeft + margin, displayTop,
+					 labelWidth, labelHeight, Qt::AlignLeft | Qt::AlignTop | Qt::TextDontClip,
+					 QString("L"));
+	painter.drawText(displayRight - margin - labelWidth, displayTop,
+					 labelWidth, labelHeight, Qt::AlignRight| Qt::AlignTop | Qt::TextDontClip,
+					 QString("R"));
+
+	// Draw the outline
+	painter.setPen(QPen(m_controls->m_colorOutline, 2, Qt::SolidLine, Qt::RoundCap, Qt::BevelJoin));
+	painter.drawRoundedRect(1, 1, width() - 2, height() - 2, 2.f, 2.f);
+
+	// Optionally measure drawing performance
+#ifdef VEC_DEBUG
+	drawTime = std::chrono::high_resolution_clock::now().time_since_epoch().count() - drawTime;
+	m_executionAvg = 0.95f * m_executionAvg + 0.05f * drawTime / 1000000.f;
+	painter.setPen(QPen(m_controls->m_colorLabels, 1, Qt::SolidLine, Qt::RoundCap, Qt::BevelJoin));
+	painter.setFont(normalFont);
+	painter.drawText(displayWidth / 2 - 50, displayBottom - 16, 100, 16, Qt::AlignLeft,
+					 QString("Exec avg.: ").append(std::to_string(m_executionAvg).substr(0, 5).c_str()).append(" ms"));
+#endif
+}
+
+
+// Periodically trigger repaint and check if the widget is visible
+void VectorView::periodicUpdate()
+{
+	m_visible = isVisible();
+	if (m_visible) {update();}
```

Comment:
According to https://doc.qt.io/qt-5/qwidget.html#update, there is no need to check visibility before calling `update()`, or am I missing something?

Reply:
This is supposed to save performance when the widget is not visible -- unlike other effects there is no need to keep processing new samples when the window is closed. I am not 100 % sure how it interacts with the `update()`, I just re-used some code that originally came from the EQ plugin, where it was specifically added to save performance. So I just _assumed_ it is somewhat important, but I did not dig deeper.